### DISCUSSION TITLE
Asking for help for ItemsControl  memory leak

### DISCUSSION THREAD

**Discussion by BonanzaBegin** [AvaloniaUI/Avalonia#11352] [Q&A] [answered]
I encountered a problem where I used ItemsControl to display a data list and bind Items to the ObservableCollection collection in the background (which limits the maximum number of data items to 500. If additional items are added later, remove the first item first and then add it again to ensure the maximum data volume is only 500) and used a data template to control the display. However, when the Window is minimized,  the process used memory is Continuously growing, is that right?

**Comment by BonanzaBegin** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/58451874/8e72ace3-2581-4910-bf7a-9712d56d63f4)
![image](https://github.com/AvaloniaUI/Avalonia/assets/58451874/8f6514b4-48ed-4d82-b968-13f975800585)

heap generation 2 is continuously growing.....

**Comment by BonanzaBegin** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/58451874/40f78acc-278c-43af-90d5-4aafe2b7f02b)

**Reply by timunie** [reply, marked as answer]
ItemsControl is not virtualized by default. If you try ListBox instead for example, memory should decrease. Rembmer to remove the outer ScrollViewer.

**Reply by timunie** [reply]
Or make ItemsControl to use VirtualizationStackPanel or use ItemsRepeater.
